src/app/clusters/thread-border-router-management-server/thread-br-mgmt-server.cpp

```diff
@@ -0,0 +1,385 @@
+    if (err == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (err != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+    ReturnOnFailure(err);
+    response.dataset = pendingDataset.AsByteSpan();
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void ServerInstance::HandleSetActiveDatasetRequest(HandlerContext & ctx,
+                                                   const Commands::SetActiveDatasetRequest::DecodableType & req)
+{
+    VerifyOrReturn(CheckFailSafeArmed(ctx));
+    VerifyOrReturn(CheckDelegate(ctx, mDelegate));
+    if (mAsyncCommandHandle.Get() != nullptr)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        ctx.SetCommandHandled();
+        return;
+    }
+
+    Thread::OperationalDataset activeDataset;
+    Thread::OperationalDataset currentActiveDataset;
+    uint64_t currentActiveDatasetTimestamp;
+    if (activeDataset.Init(req.activeDataset) != CHIP_NO_ERROR)
+    {
+        // If any of the parameters in the ActiveDataset is invalid, the command SHALL fail with a status code
+        // of INVALID_COMMAND.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
+
+    if (mDelegate->GetActiveDataset(currentActiveDataset) == CHIP_NO_ERROR &&
+        currentActiveDataset.GetActiveTimestamp(currentActiveDatasetTimestamp) == CHIP_NO_ERROR)
+    {
+        // If this command is invoked when the ActiveDatasetTimestamp attribute is not null, the command SHALL
+        // fail with a status code of INVALID_IN_STATE.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidInState);
+        return;
+    }
+
+    mPath               = ctx.mRequestPath;
+    mAsyncCommandHandle = CommandHandler::Handle(&ctx.mCommandHandler);
+    ctx.mCommandHandler.FlushAcksRightAwayOnSlowCommand();
+    mBreadcrumb    = req.breadcrumb;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    if ((err = mDelegate->SetActiveDataset(activeDataset, this)) != CHIP_NO_ERROR)
+    {
+        OnActivateDatasetComplete(err);
+    }
+}
+
+void ServerInstance::HandleSetPendingDatasetRequest(HandlerContext & ctx,
+                                                    const Commands::SetPendingDatasetRequest::DecodableType & req)
+{
+    VerifyOrReturn(CheckDelegate(ctx, mDelegate));
+    bool panChangeSupported;
+    if (mDelegate->GetPanChangeSupported(panChangeSupported) != CHIP_NO_ERROR || !panChangeSupported)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::UnsupportedCommand);
+        return;
+    }
+    Thread::OperationalDataset pendingDataset;
+    if (pendingDataset.Init(req.pendingDataset) != CHIP_NO_ERROR)
+    {
+        // If any of the parameters in the PendingDataset is invalid, the command SHALL fail with a status code
+        // of INVALID_COMMAND.
```

Comment:
The spec says nothing like this.  I believe that it means to, since it has an "Otherwise" that is not connected to anything, but right now it just does not say.  Spec issue needed.

Reply:
Filed the issue https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/9391